save-backend/src/main/kotlin/com/saveourtool/save/backend/service/TestSuitesSourceService.kt

```diff
@@ -151,12 +153,45 @@ class TestSuitesSourceService(
         testSuitesSource: TestSuitesSourceDto,
         mode: TestSuitesSourceFetchMode,
         version: String,
-    ): Mono<EmptyResponse> = preprocessorWebClient
-        .post()
-        .uri("/test-suites-sources/fetch?mode={mode}&version={version}", mode, version)
-        .bodyValue(testSuitesSource)
-        .retrieve()
-        .toBodilessEntity()
+    ): Mono<EmptyResponse> = testsSourceVersionService.doesContain(
+        organizationName = testSuitesSource.organizationName,
+        sourceName = testSuitesSource.name,
+        version = version,
+    )
+        .filterWhen { doesContain ->
+            if (doesContain && mode == TestSuitesSourceFetchMode.BY_BRANCH) {
+                log.debug {
+                    "Detected that source ${testSuitesSource.organizationName}/${testSuitesSource.name} already contains such version $version and it should be" +
+                            " overridden."
+                }
+                testsSourceVersionService.delete(
+                    organizationName = testSuitesSource.organizationName,
+                    sourceName = testSuitesSource.name,
+                    version = version,
+                )
+                    .filter { it }
+                    .switchIfEmptyToResponseException(HttpStatus.INTERNAL_SERVER_ERROR) {
+                        "Failed to delete existed version $version in ${testSuitesSource.organizationName}/${testSuitesSource.name}"
+                    }
+                    .thenReturn(true)
+            } else if (doesContain) {
+                log.debug {
```

Comment:
may be move into separate variable this msg? seems, it duplicates

Reply:
will extract to a dedicated method